File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int max = 0;
        int left = 0;
        int right = heights.length-1;
        int l = heights.length -1;

        while (left < right){
            int min = Math.min(heights[left], heights[right]);
            int area = min * l;
            max = Math.max(max, area);
            if (heights[left]<=heights[right]){
                left ++;
            }else {
                right --;
            }
            l--;
        }
        return max;
    }
}

File: Data Structures & Algorithms/maximum-score-after-splitting-a-string/submission-0.py
class Solution:
    def maxScore(self, s: str) -> int:
        left = 0
        right = 0
        max_count = 0

        for num in s : 
            if num=="1":
                right += 1
        
        for i in range(len(s)-1) : 
            if s[i] == "0":
                left += 1 
                max_count = max(max_count, left+right)
            else : 
                right -= 1
                max_count = max(max_count, left+right)

        return max_count
